### PR TITLE
Update Program.cs

### PR DESCRIPTION
**Pull request by davidfowl**
Use model binding

# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] The quickstart code compiles correctly
* [ ] You've tested new builds of the quickstart if you changed quickstart code
* [ ] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: service_invocation/csharp/http/order-processor/Program.cs [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+if (app.Environment.IsDevelopment())
+{
+    app.UseDeveloperExceptionPage();
+}
 
-app.MapPost("/orders", async context => {
-    var data = await context.Request.ReadFromJsonAsync<Order>();
-    Console.WriteLine("Order received : " + data);
-    await context.Response.WriteAsync(data.ToString());
+app.MapPost("/orders", (Order order) =>
```
Comment (paulyuk) [inline service_invocation/csharp/http/order-processor/Program.cs:10]: Thank you @davidfowl!.  This is a nice improvement when json properties map identically.


Q: can and should we show this Async?

Comment (yaron2) [reply service_invocation/csharp/http/order-processor/Program.cs:10]: ping @davidfowl

**Review by amulyavarote — APPROVED** [review]

File: service_invocation/csharp/http/order-processor/Program.cs (hunk 1 above)
Comment (davidfowl) [reply service_invocation/csharp/http/order-processor/Program.cs:10]: No need to do async here. That happens before the method is called with the order.

**Review by amulyavarote — APPROVED** [review]

**Comment by amulyavarote** [comment]
@davidfowl Can you please take the recent changes from master and rebase this with master and push the changes?

**Pull request merged**